src/test/java/com/list/todo/services/UserServiceTest.java

```diff
@@ -0,0 +1,242 @@
+        //arrange
+        String partOfUsername = "us";
+        List<User> users = new ArrayList<>();
+        User user1 = createUser(1);
+        User user2 = createUser(2);
+        users.add(user1);
+        users.add(user2);
+        Set<String> usernames = new HashSet<>();
+        usernames.add(user1.getUsername());
+        usernames.add(user2.getUsername());
+
+        when(userRepository.findByUsernameLike(partOfUsername + "%")).thenReturn(users);
+
+        //act
+        Set<String> returnedUsernames = userService.searchUsersByPartOfUsername(partOfUsername);
+
+        //assert
+        verify(userRepository).findByUsernameLike(partOfUsername + "%");
+        Assert.assertEquals(usernames, returnedUsernames);
+    }
+
+    @Test
+    public void updateUser_OnExistentUser_ReturnsAnObjectOfUser() {
+        //arrange
+        User user = createUser(1);
+        User updatedUser = createUser(2);
+        UpdatingUserInput userInput = new UpdatingUserInput();
+        userInput.setName(updatedUser.getName());
+        userInput.setUsername(updatedUser.getUsername());
+        userInput.setEmail(updatedUser.getEmail());
+        userInput.setPassword(updatedUser.getPassword());
+        user = spy(user);
+
+        when(userRepository.findById(currentUserId)).thenReturn(Optional.of(user));
+        when(bCryptPasswordEncoder.encode(updatedUser.getPassword())).thenReturn(updatedUser.getPassword());
+        when(userRepository.save(user)).thenReturn(updatedUser);
+
+        //act
+        Optional<User> returnedUser = userService.updateUser(currentUserId, userInput);
+
+        //assert
+        verify(user).setName(userInput.getName());
+        verify(user).setUsername(userInput.getUsername());
+        verify(user).setEmail(userInput.getEmail());
+        verify(user).setPassword(userInput.getPassword());
+        Assert.assertEquals(updatedUser, returnedUser.get());
+    }
+
+    @Test
+    public void deleteUser_OnExistentUser_ReturnsVoid() {
+        //arrange
+        User user = createUser(1);
+        List<Follower> followers = new ArrayList<>();
+        followers.add(new Follower());
+        followers.add(new Follower());
+        List<Follower> followedUsers = new ArrayList<>();
+        followedUsers.add(new Follower());
+        followedUsers.add(new Follower());
+        List<TodoList> todoLists = new ArrayList<>();
+        todoLists.add(new TodoList());
+        todoLists.add(new TodoList());
+
+        when(userRepository.findById(currentUserId)).thenReturn(Optional.of(user));
+        when(followerRepository.findByFollower(user)).thenReturn(followedUsers);
+        when(followerRepository.findByFollowedUserId(currentUserId)).thenReturn(followers);
+        when(todoListRepository.findByCreatedBy(user.getUsername())).thenReturn(todoLists);
+
+        //act
+        userService.deleteUser(currentUserId);
+
+        //assert
+        verify(userRepository).findById(currentUserId);
```

Comment:
Does order of methods invocation matter in this case? If yes, use mockito InOrder to verify it

Reply:
It doesnt matter